### PR TITLE
ci: regroup e2e CIs and add CI to test LTS cycle first SDK

### PR DESCRIPTION
**Pull request by Jonas1312**
- put pyright job in ci.yml
- regrouped all e2e tests workflow into one workflow, with different test modes
- we can now choose to run notebooks and/or e2e tests when triggering manually
- we also have a test mode "ALL" when triggering manually, to run all e2e test modes
- new CI "LTS_CYCLE_FIRST_SDK" that tests the first SDK of the cycle against staging. The SDK version is hardcoded in the workflow. Maybe not the best idea since we might forget to update it...?

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -1,47 +1,183 @@
...
+        description: "Test mode: MASTER_STAGING (master branch against staging), RELEASE_PREPROD (latest release branch against preprod), MASTER_LTS (master branch against LTS), PREV_RELEASE_TESTS_STAGING (SDK from master, previous release tests against staging), LTS_CYCLE_FIRST_SDK (test with first SDK of LTS cycle, against staging)"
+        required: true
+        type: choice
+        default: "MASTER_STAGING"
+        options:
+          - MASTER_STAGING
+          - RELEASE_PREPROD
+          - MASTER_LTS
+          - PREV_RELEASE_TESTS_STAGING
+          - LTS_CYCLE_FIRST_SDK
```
Comment (theodu) [inline .github/workflows/e2e_tests.yml:32]: can we have the option to launch all of them? I feel that it would be the done to use the most

**Review by theodu — APPROVED** [review]
It looks good to me.

File: .github/workflows/e2e_tests.yml (hunk 1 above)
Comment (Jonas1312) [reply .github/workflows/e2e_tests.yml:32]: I can try to implement that yep

Comment (Jonas1312) [reply .github/workflows/e2e_tests.yml:34]: done:  https://github.com/kili-technology/kili-python-sdk/actions/runs/4305553000

File: .github/workflows/e2e_tests.yml [hunk 2]
```diff
@@ -1,47 +1,205 @@
...
+        required: true
+        type: choice
+        default: "MASTER_STAGING"
+        options:
+          - ALL
+          - MASTER_STAGING
+          - RELEASE_PREPROD
+          - MASTER_LTS
+          - PREV_RELEASE_TESTS_STAGING
+          - LTS_CYCLE_FIRST_SDK
```
Comment (PierreLeveau) [inline .github/workflows/e2e_tests.yml:34]: LTS_CYCLE_FIRST_SDK_STAGING

File: .github/workflows/e2e_tests.yml [hunk 3]
```diff
@@ -1,47 +1,205 @@
...
   push:
     branches:
       - "release/**"
       - master
   schedule:
-    - cron: "0 7,15 * * 1-5" # 8am and 4pm UTC+1 every weekday
+    - cron: "0 7,15 * * 1-5" # MASTER_STAGING
+    - cron: "0 10 * * 1-5" # MASTER_LTS
+    - cron: "0 12 * * 1-5" # PREV_RELEASE_TESTS_STAGING
+    - cron: "0 14 * * 1-5" # LTS_CYCLE_FIRST_SDK
```
Comment (PierreLeveau) [inline .github/workflows/e2e_tests.yml:43]: LTS_CYCLE_FIRST_SDK_STAGING

Comment (PierreLeveau) [reply .github/workflows/e2e_tests.yml:43]: (here and elsewhere)

**Review by PierreLeveau — APPROVED** [review]

**Comment by Jonas1312** [comment]
image

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**